### PR TITLE
Add a simple table for functions to the functional module

### PR DESCRIPTION
**Pull request by kuettler**

### CODE REVIEW

File: std/functional.d [hunk 1]
```diff
@@ -3,6 +3,49 @@
...
+        function's argument.
+    ))
+    $(TR $(TD $(D $(LREF memoize)))
+        $(TD Creates a function that caches its result for fast reevalation.
+    ))
+    $(TR $(TD $(D $(LREF toDelegate)))
+        $(TD Converts a callable to a delegate.
+    ))
+    $(TR $(TD $(D $(LREF forward)))
+        $(TD Forwards function arguments with saving ref-ness.
```
Comment (Hackerpilot) [inline std/functional.d]: Suggestion: "while" instead of "with"

File: std/functional.d [hunk 2]
```diff
@@ -3,6 +3,49 @@
...
+        $(TD Joins a couple of functions into one that executes the original
+        functions independently and returns a tuple with all the results.
+    ))
+    $(TR $(TD $(D $(LREF compose)), $(D $(LREF pipe)))
+        $(TD Join a couple of functions into one that executes the original
+        functions one after the other, using one function's result for the next
+        function's argument.
+    ))
+    $(TR $(TD $(D $(LREF memoize)))
+        $(TD Creates a function that caches its result for fast reevalation.
```
Comment (Hackerpilot) [inline std/functional.d]: "re-evaluation"?

File: std/functional.d [hunk 3]
```diff
@@ -3,6 +3,49 @@
...
+$(LINK2 std_algorithm.html, std.algorithm) or $(LINK2 std_range.html,
+std.range).
+
+$(BOOKTABLE ,
+    $(TR $(TD $(D $(LREF unaryFun)), $(D $(LREF binaryFun)))
+        $(TD Create a unary or binary function from a string. Most often
+        used when defining algorithms on ranges.
+    ))
+    $(TR $(TD $(D $(LREF lessThan)), $(D $(LREF greaterThan)), $(D $(LREF equalTo)))
+        $(TD Ready made predicate functions to compare two values.
```
Comment (Hackerpilot) [inline std/functional.d]: "that compares" instead of "to compare" would make this more consistent with the other descriptions.

Comment (kuettler) [inline std/functional.d]: Plural. Three functions.

Comment (quickfur) [inline std/functional.d]: Note that in English, "ready-made" should be hyphenated.

**Comment by quickfur** [comment]
Should the list be sorted, for easier human searching?

**Comment by quickfur** [comment]
Other than that, it's looking good. This is a much-needed doc improvement. Thanks!!

**Comment by kuettler** [comment]
Yes, I refrained from sorting since there are groups of functions, but it really makes a difference. You are right.

**Comment by quickfur** [comment]
If there are groups of functions, probably we should group them by category, and sort them within each group? E.g., like the old std.algorithm docs here: http://dlang.org/phobos/std_algorithm.html

**Comment by kuettler** [comment]
Nah, too much grouping. This is a small module with a short list of functions, some of which are closely related. Like `lessThan` and `greaterThan`. I am aiming for a welcoming summery that gives the curios visitor a first impression. Putting `lessThan` and `greaterThan` on the same line matches everyone's thinking (at least I hope so). This is not auto-generated and it is not meant to pretend to be.

**Comment by quickfur** [comment]
Fair enough. Let's merge, we can improve on this later if necessary.

**Comment by quickfur** [comment]
Auto-merge toggled on

**Comment by kuettler** [comment]
Thank you. :)

**Pull request merged**